LokiPackage/Sources/Core/UserDefaults/UserDefaultsClient.swift

```diff
@@ -1,14 +1,16 @@
 import Foundation
 
-package protocol UserDefaultsClient {
+package protocol UserDefaultsClient: Sendable {
     func object<V: Decodable>(forKey key: UserDefaultsKey) throws -> V
     func set<V: Encodable>(_ value: V, forKey key: UserDefaultsKey) throws
 }
 
-package final class DefaultUserDefaultsClient {
+package final class DefaultUserDefaultsClient: @unchecked Sendable {
```

Comment:
L13 で `nonisolated(unsafe)` を付けていると `@unchecked Sendable` は不要だと思ったんだけど、付けないと以下のエラーが発生する。

```
Stored property 'userDefaults' of 'Sendable'-conforming class 'DefaultUserDefaultsClient' has non-sendable type 'UserDefaults'
```

Reply:
そもそも `nonisolated(unsafe)` が何をしているか正確には理解できていない。
あと付けるとビルド時に以下の警告が発生するが、Swift 5.10 のコンパイラに不具合があるのか、私のコードが悪いのかわかっていない。

```
warning: Skipping correcting file because it produced Swift parser errors: /Users/uhooi/ghq/github.com/uhooi/Loki/LokiPackage/Sources/Core/Log/Logger.swift
{
  "diagnostics" : [
    "unexpected code '(unsafe)' before modifier"
  ]
}
```

Reply:
今回の変更で追加した`Sendable`を全て剥がして、`shared`の部分に`nonisolated(unsafe)`をつけると（warningが出る）問題自体は解決します！

ただ、@yuukiw00wさんがおっしゃっているように`DefaultUserDefaultsClient`自体を`Sendable`にできたほうが幸せだと思っています！！

Reply:
@ojun9 まじ？ でも `Logger` は `Sendable` に準拠してないけど↑とほぼ同じ警告が発生したままだよ、、 https://github.com/uhooi/Loki/pull/224/files#diff-b0b484acfa47e1abf56e3f74523e485ff1361112486f6d3ffff42e8a3afeb41bR7

Reply:
あれれ、僕の手元では`Logger`の部分も警告は出てないですね😢


```shell
$ swift -version
swift-driver version: 1.90.11.1 Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
Target: arm64-apple-macosx14.0
```

Reply:
`SwiftLint`を実行すると↓の`warning`が出現しそうなので、`SwiftLint`側が`(unsafe)`キーワードに対応してないのが原因な気がしてきました

```shell
warning: Skipping correcting file because it produced Swift parser errors: /Users/uhooi/ghq/github.com/uhooi/Loki/LokiPackage/Sources/Core/Log/Logger.swift
{
  "diagnostics" : [
    "unexpected code '(unsafe)' before modifier"
  ]
}
```

Reply:
あああ…これ SwiftLint の警告だった、、失礼しましたぁ、、、